File: algorithmProblemSolving/programmers/level1/모의고사.js
function solution(answers) {
    var answer = [];
    var spj1 = [1,2,3,4,5];
    var spj2 = [2,1,2,3,2,4,2,5];
    var spj3 = [3,3,1,1,2,2,4,4,5,5];
    var score = [0,0,0];
    for(var i=0;i<answers.length;i++){
        if(answers[i]==spj1[i%spj1.length]) score[0]++;
        if(answers[i]==spj2[i%spj2.length]) score[1]++;
        if(answers[i]==spj3[i%spj3.length]) score[2]++;
    }
    var max = score.reduce(function(a,b){
        return a>b?a:b;
    });
    for(var i=0;i<score.length;i++){
        if(score[i]==max) answer.push(i+1);
    }
    return answer;
}

File: algorithmProblemSolving/programmers/level1/모의고사.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;

vector<int> solution(vector<int> answers) {
	vector<int> answer;
	vector<vector<int>>supo = { { 1,2,3,4,5 },{ 2, 1, 2, 3, 2, 4, 2, 5 },{ 3, 3, 1, 1, 2, 2, 4, 4, 5, 5 } };
	vector<int> score;
	
	int tmp_s = 0;
	for (int i = 0; i < supo.size(); i++) {
		tmp_s = 0;
		for (int j = 0; j < answers.size(); j++) {
			if (answers[j] == supo[i][j%supo[i].size()]) tmp_s++;
		}
		score.push_back(tmp_s);
	}
	vector<int> max_v = score;
	sort(score.begin(), score.end());
	int max = score[score.size() - 1];
	for (int i = 0; i < score.size(); i++) {
		if (max_v[i] == max) answer.push_back(i + 1);
	}

	return answer;
}